window_manager/src/XWindow.cpp

```diff
@@ -0,0 +1,213 @@
+#include "XWindow.hpp"
+
+#include <future>
+
+GLXFBConfig* chooseFbConfig(Display* d, int screen) {
+    int glAttrib[] = {
+            GLX_DRAWABLE_TYPE , GLX_WINDOW_BIT,
+            GLX_RENDER_TYPE   , GLX_RGBA_BIT,
+            GLX_RED_SIZE      , 1,
+            GLX_GREEN_SIZE    , 1,
+            GLX_BLUE_SIZE     , 1,
+            GLX_ALPHA_SIZE    , 8,
+            GLX_DEPTH_SIZE    , 24,
+            GLX_DOUBLEBUFFER  , True,
+            None
+    };
+    int num;
+    GLXFBConfig * ret = glXChooseFBConfig(d, screen, glAttrib, &num);
+    if (ret == nullptr)
+        throw std::runtime_error("Unable to find a frame buffer config!");
+
+    return ret;
+}
+
+namespace rcbe::core {
+XWindow::XWindow(WindowConfig &&config, Display* root_display, int screen_number, const Window& root_window, const Atom& delete_msg)
+:
+config_ { std::move(config) }
+, root_display_(root_display)
+, rendering_context_ { (config_.rendering_window) ? std::make_shared<rendering::RenderingContext>() : nullptr }
+{
+    GLXFBConfig* fbConfigs = chooseFbConfig(root_display_, screen_number);
+    XVisualInfo *visInfo = glXGetVisualFromFBConfig(root_display_, fbConfigs[0]);
+
+    rendering_context_->x_display = XOpenDisplay(NULL);
+    rendering_context_->background_color = config_.background_color;
+    rendering_context_->window_dimensions = config.size;
+    rendering_context_->x_delete_message = delete_msg;
+
+    attributes_.colormap = XCreateColormap(rendering_context_->x_display, root_window, visInfo->visual, AllocNone);
+    rendering_context_->gl_x_window = XCreateWindow(rendering_context_->x_display, root_window,
```

Comment:
change this depending on enum type of window

Reply:
This system needs a lot of refactoring to actually support various Windows and goes out of scope of this task.